lib/src/testing_library/dom/async/wait_for.dart

```diff
@@ -0,0 +1,151 @@
+/// https://testing-library.com/docs/dom-testing-library/api-async/#waitfor
+@JS()
+library over_react_test.src.testing_library.dom.async.wait_for;
+
+import 'dart:async';
+import 'dart:html' show Element, MutationObserver, document, promiseToFuture;
+import 'dart:js' show allowInterop;
+
+import 'package:js/js.dart';
+import 'package:test/test.dart';
+
+import 'package:over_react_test/src/testing_library/dom/async/types.dart';
+import 'package:over_react_test/src/testing_library/dom/config/configure.dart' show getConfig;
+
+export 'package:over_react_test/src/testing_library/dom/async/types.dart' show MutationObserverInit;
+
+/// Calls the provided [expectation] on a given [interval] and/or when the [container] DOM changes,
+/// completing only if it does not throw a [TestFailure], or by throwing that [TestFailure] if
+/// the [timeout] expires before the [expectation] succeeds.
+///
+/// Similar to <https://testing-library.com/docs/dom-testing-library/api-async/#waitfor>, but designed to
+/// work with the Dart test package's [expect] function(s).
+///
+/// * If you're waiting for an element to existing in the DOM, using a `findBy*` query instead.
+/// * If you're waiting for an element to be removed from the DOM, use [waitForElementToBeRemoved] instead.
+///
+/// * [timeout] defaults to 1000ms, or whatever value has been set globally for `Config.asyncUtilTimeout`.
+/// * [interval] defaults to 50ms.
+/// * [onTimeout] can be used to customize the [TestFailure] message.
+Future<T> waitFor<T>(
```

Comment:
I ended up writing our own dart function so that dart authors could use dart test `expect`s as the `expectation` argument.

The logic in the function is a 1:1 match of how the JS function is written, and works exactly the same.